api/src/main/java/com/messagebird/MessageBirdServiceImpl.java

```diff
@@ -213,6 +236,58 @@ public void deleteByID(String request, String id) throws UnauthorizedException,
         }
     }
 
+    /**
+     * By default, HttpURLConnection does not support PATCH requests. We can
+     * however work around this with reflection. Many thanks to okutane on
+     * StackOverflow: https://stackoverflow.com/a/46323891/3521243.
+     */
+    private synchronized static void allowPatchRequestsIfNeeded() throws GeneralException {
+        if (isPatchRequestAllowed) {
+            // Don't do anything if we've run this method before. We're in a
+            // synchronized block, so return ASAP.
+            return;
+        }
+
+        try {
+            // Ensure we can access the fields we need to set.
+            Field methodsField = HttpURLConnection.class.getDeclaredField("methods");
+            methodsField.setAccessible(true);
+
+            Field modifiersField = Field.class.getDeclaredField("modifiers");
+            modifiersField.setAccessible(true);
+            modifiersField.setInt(methodsField, methodsField.getModifiers() & ~Modifier.FINAL);
+
+            // Determine what methods should be allowed.
+            String[] allowedMethods = getAllowedMethods((String[]) methodsField.get(null));
```

Comment:
I like the `noInstanceBecauseStaticField` you used for `methodsField.set(noInstanceBecauseStaticField, allowedMethods);` maybe you could do the same there with `methodsField.get(noInstanceBecauseStaticField)`?

Reply:
Resolved in https://github.com/messagebird/java-rest-api/pull/44/commits/bf8d507a7f05660ff7ad01af53c11810c5949f7f